lib/web_ui/lib/src/engine/text_editing.dart

```diff
@@ -583,8 +625,19 @@ class HybridTextEditing {
   /// Also used to define if a keyboard is needed.
   bool _isEditing = false;
 
-  /// Flag indicating if the flutter framework requested a keyboard.
-  bool get needsKeyboard => _isEditing;
+  /// Flag indication the input element needs to be positioned.
+  ///
+  /// See [TextEditingElement._delayBeforePositining].
+  bool get inputElementNeedsToBePositioned =>
+      !inputPositioned &&
+      _isEditing &&
+      browserEngine == BrowserEngine.webkit &&
+      operatingSystem == OperatingSystem.iOs;
+
+  /// Flag indication wheterher the input element's position is set.
+  ///
+  /// See [inputElementNeedsToBePositioned].
+  bool inputPositioned = false;
```

Comment:
It feels like many of the fields here really belong to the `TextEditingElement` class (e.g. `inputPositioned`, `inputElementNeedsToBePositioned`, `_editingLocationAndSize`, etc).

Reply:
Thanks for the suggestion. I agree, I'll make a refactor in the next PR.